src/main/java/nearsoft/skytouch/task/controller/ProductController.java

```diff
@@ -0,0 +1,59 @@
+package nearsoft.skytouch.task.controller;
+
+import nearsoft.skytouch.task.model.Product;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.ui.ModelMap;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+@Controller
+public class ProductController {
+
+    @RequestMapping(value= "/view/product", method = RequestMethod.GET)
+    public String addProducts(Map<String, Object> model) {
+        model.put("product",new Product());
+        return "/products/newProduct";
+    }
+
+    @RequestMapping(value="/products", method = RequestMethod.POST)
+    public String submit(@ModelAttribute("product")Product product,
+                         BindingResult bindingResult,Map<String, Object> model) {
+        if (bindingResult.hasErrors()) {
+            model.put("errors",bindingResult.getAllErrors());
+            return "/errors/error";
+        }
+        return "/products/newProduct";
+    }
+
+    @RequestMapping(value="/products", method=RequestMethod.GET)
+    public String listProducts(Map<String, Object> model){
+        Product product;
```

Comment:
I understand you are hardcoding this because the micro service is not ready yet... but I suggest to leave the controller cleaner by using some service interface... later your service or repository class  is going to connect to the queue but without the need of modifying existing code (open-close principle).

Reply:
Ok i'm working on that